### DISCUSSION TITLE
"Alias" Gdiffsplit

### DISCUSSION THREAD

**Discussion by myitcv** [tpope/vim-fugitive#1899] [Q&A] [answered]
Firstly, thank you for all your excellent Vim plugins, @tpope. And to the community who are helping to answer questions like this.

Following on from #1894, in particular a question regarding https://github.com/tpope/vim-fugitive/commit/d523feebe99bb93a9309ef83cbac2b71ac408c36

At current tip, to get the behaviour I previously had in 8e4a677 and before, I now need to call:

```
:leftabove Gdiffsplit! [arg]
```

whereas previously I was able to simply do:

```
:Gdiff [arg]
```

_(no complaints mind, the new form is much more precise)_

Is there a way to "alias" a command call in this way?

Thanks in advance

**Comment by tpope** [comment, marked as answer]
This ought to do it:

`command! -bar -bang -nargs=* GDiff Gdiffsplit<bang> <args>`

Nothing changed regarding `!` afaik, so I've abstained from hard-coding it.

**Reply by myitcv** [reply]
Wonderful, thank you!

For completeness' sake, I did the following:

```
command! -bar -bang -nargs=* -complete=customlist,fugitive#EditComplete Gdiff Gvdiffsplit! <args>
```

* Named `Gdiff` to intentionally collide with `Gdiffsplit` so that tab completing chooses my variant
* Provided the `-complete` flag so that I can enjoy the tab completion of arguments, when supplied, to my `Gdiff` variant (e.g. table completion of ref names likes branches)
* Hardcode the use of `!` on the RHS of the command definition, because that's what I always want

> Nothing changed regarding `!` afaik, so I've abstained from hard-coding it.

Agreed. It's more that I've only just found out about the `!` variant so started using it!
